File: VB/Scaffolding.DetailCollections.EFCore/Migrations/DepartmentContextModelSnapshot.vb
Imports System
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Infrastructure
Imports Microsoft.EntityFrameworkCore.Metadata
Imports Microsoft.EntityFrameworkCore.Migrations
Imports Scaffolding.DetailCollectionsEFCore.Model

Namespace Scaffolding.DetailCollections.EFCore.Migrations
	<DbContext(GetType(DepartmentContext))>
	Partial Friend Class DepartmentContextModelSnapshot
		Inherits ModelSnapshot

		Protected Overrides Sub BuildModel(ByVal modelBuilder As ModelBuilder)
			modelBuilder.HasAnnotation("ProductVersion", "1.0.1").HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn)

			modelBuilder.Entity("Scaffolding.DetailCollections.Model.Course", Sub(b)
				b.Property(Of Integer)("CourseID").ValueGeneratedOnAdd()
				b.Property(Of Integer)("DepartmentID")
				b.Property(Of String)("Title").IsRequired()
				b.HasKey("CourseID")
				b.HasIndex("DepartmentID")
				b.ToTable("Courses")
			End Sub)

			modelBuilder.Entity("Scaffolding.DetailCollections.Model.Department", Sub(b)
				b.Property(Of Integer)("DepartmentID").ValueGeneratedOnAdd()
				b.Property(Of String)("Name")
				b.HasKey("DepartmentID")
				b.ToTable("Departments")
			End Sub)

			modelBuilder.Entity("Scaffolding.DetailCollections.Model.Employee", Sub(b)
				b.Property(Of Integer)("EmployeeID").ValueGeneratedOnAdd()
				b.Property(Of Integer)("DepartmentID")
				b.Property(Of String)("Name")
				b.HasKey("EmployeeID")
				b.HasIndex("DepartmentID")
				b.ToTable("Employees")
			End Sub)

			modelBuilder.Entity("Scaffolding.DetailCollections.Model.Course", Sub(b) b.HasOne("Scaffolding.DetailCollections.Model.Department", "Department").WithMany("Courses").HasForeignKey("DepartmentID").OnDelete(DeleteBehavior.Cascade))

			modelBuilder.Entity("Scaffolding.DetailCollections.Model.Employee", Sub(b) b.HasOne("Scaffolding.DetailCollections.Model.Department", "Department").WithMany("Employees").HasForeignKey("DepartmentID").OnDelete(DeleteBehavior.Cascade))
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.DetailCollections.EFCore/Migrations/DepartmentContextModelSnapshot.cs
using System;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Infrastructure;
using Microsoft.EntityFrameworkCore.Metadata;
using Microsoft.EntityFrameworkCore.Migrations;
using Scaffolding.DetailCollectionsEFCore.Model;

namespace Scaffolding.DetailCollections.EFCore.Migrations {
    [DbContext(typeof(DepartmentContext))]
    partial class DepartmentContextModelSnapshot : ModelSnapshot {
        protected override void BuildModel(ModelBuilder modelBuilder) {
            modelBuilder
                .HasAnnotation("ProductVersion", "1.0.1")
                .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);

            modelBuilder.Entity("Scaffolding.DetailCollections.Model.Course", b =>
                {
                    b.Property<int>("CourseID")
                        .ValueGeneratedOnAdd();

                    b.Property<int>("DepartmentID");

                    b.Property<string>("Title")
                        .IsRequired();

                    b.HasKey("CourseID");

                    b.HasIndex("DepartmentID");

                    b.ToTable("Courses");
                });

            modelBuilder.Entity("Scaffolding.DetailCollections.Model.Department", b =>
                {
                    b.Property<int>("DepartmentID")
                        .ValueGeneratedOnAdd();

                    b.Property<string>("Name");

                    b.HasKey("DepartmentID");

                    b.ToTable("Departments");
                });

            modelBuilder.Entity("Scaffolding.DetailCollections.Model.Employee", b =>
                {
                    b.Property<int>("EmployeeID")
                        .ValueGeneratedOnAdd();

                    b.Property<int>("DepartmentID");

                    b.Property<string>("Name");

                    b.HasKey("EmployeeID");

                    b.HasIndex("DepartmentID");

                    b.ToTable("Employees");
                });

            modelBuilder.Entity("Scaffolding.DetailCollections.Model.Course", b =>
                {
                    b.HasOne("Scaffolding.DetailCollections.Model.Department", "Department")
                        .WithMany("Courses")
                        .HasForeignKey("DepartmentID")
                        .OnDelete(DeleteBehavior.Cascade);
                });

            modelBuilder.Entity("Scaffolding.DetailCollections.Model.Employee", b =>
                {
                    b.HasOne("Scaffolding.DetailCollections.Model.Department", "Department")
                        .WithMany("Employees")
                        .HasForeignKey("DepartmentID")
                        .OnDelete(DeleteBehavior.Cascade);
                });
        }
    }
}
